src/hash.cr

```diff
@@ -272,6 +272,26 @@ class Hash(K, V)
     self
   end
 
+  def select(&block : K, V -> U)
+    inject({} of K => V) do |memo, k, v|
+      yield(k, v) ? memo.merge!({k => v}) : memo
```

Comment:
`each_with_object` should allow to get rid of the ternary in favor of a modifier-if.